### DISCUSSION TITLE
Assignees from CODEOWNERS not working

### DISCUSSION THREAD

**Discussion by aldenquimby** [renovatebot/renovate#18194] [Request Help] [answered]
I recently spent time debugging an issue with `assigneesFromCodeOwners` not working on Github.

Google landed me on https://github.com/renovatebot/config-help/issues/913

Since I can't comment on that old issue, I'm creating a Q&A entry so that I can document the solution I found for anyone else that stumbles across the old issue

**Steps to reproduce issue**
1. `renovate.json`:
```json
{
  "assigneesFromCodeOwners": true
}
```
2. `CODEOWNERS`:
```
/my-project-within-monorepo/package.json @aldenquimby
```
3. renovate PR upgrades a dependency in `/my-project-within-monorepo/package.json`

**Expected**
- `@aldenquimby` tagged as both reviewer and assignee on PR (reviewer happens automatically via Github)

**Actual**
- `@aldenquimby` tagged as reviewer, but not assignee on PR

**Comment by aldenquimby** [comment, marked as answer]
**How did we fix it?**
Change `CODEOWNERS` to this:
```
/my-project-within-monorepo/package*.json @aldenquimby
```

**Why does that fix it?**
- renovate PRs modify both `package.json` and `package-lock.json`
- renovate [requires _every_ changed file in the PR to match `CODEOWNERS`](https://github.com/renovatebot/renovate/blob/3bbd8b891445f5ae883e2ddb3320f1a1e2cd9554/lib/workers/repository/update/pr/code-owners.ts#L44)

**How else could you fix it?**
Change `CODEOWNERS` to this:
```
/my-project-within-monorepo/ @aldenquimby
```
Warning - Github will start adding you as a reviewer on PRs that change any file in your project, which you may not want

**Comment by rarkins** [comment]
Sounds like we could/should fix this so that we adding if the user matched any file, not all?
